internal/store/utils.go

```diff
@@ -169,7 +169,7 @@ func isNativeResource(name v1.ResourceName) bool {
 }
 
 func isPrefixedNativeResource(name v1.ResourceName) bool {
-	return strings.Contains(string(name), v1.ResourceDefaultNamespacePrefix)
+	return false
 }
```

Comment:
If we're just going to return false, then we should just get rid of this method.

Reply:
Also there should be a test for this, we shouldn't just be able to change this function without some test breaking.

Reply:
I think we should remove the code that check whether the resource is native or not as a whole.

The idea would be to make `Integer` the default unit if nothing is matched after https://github.com/kubernetes/kube-state-metrics/blob/d3afe69fd784b7ea31a42ce2eaceb7f7b1ead7a0/internal/store/pod.go#L193-L204 and remove the condition for extended resources: https://github.com/kubernetes/kube-state-metrics/blob/d3afe69fd784b7ea31a42ce2eaceb7f7b1ead7a0/internal/store/pod.go#L205-L211

Reply:
updated, i noticed there is already a test case for "kubernetes.io/resource-foo" so just updated expectVal